### ISSUE TITLE
Read and Load IPFS Snapshot upon worker start up

### ISSUE THREAD

**Issue by pavankanteti** [Polkadex-Substrate/polkadexTEE-worker#205]

**Comment by haerdib** [comment]
In case the permanent storage is damaged or a failover worker takes over, the state needs to be recovered from an IPFS snapshot.  To ensure that the permanent storage is up  to date, upon start up a worker should check the IPFS snapshot and compare it to its own state. If the disc storage is outdated, IPFS snapshot will be loaded and disc storage is updated. 

This applies to both nonce and balance storage

Checkout https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/develop/worker/src/ipfs.rs for ipfs interaction example

**Comment by haerdib** [comment]
@Gauthamastro to be able to see which storage is more recent, we could store the blocknumber (guess that doens't work as we're moving balances also offline..?) or a timestamp together with it or something like that .. or do you have a different idea?

**Comment by pavankanteti** [comment]
Can be estimated after #214

**Comment by haerdib** [comment]
IPFS storage (see issue #224) is read and permanent storage (disc) is updated. 
Unit testing: can not be done for IPFS reading, but the storing / writing to db can be unit tested
What will probably take some time : connecting to external IPFS server, connection issues.

Not part of this issue but as an explanation (and documentation):
Enclave inmemory will be updated then by reading permanent storage afterwards (which is already implemented, atleast for orderbook mirror).

**Comment by haerdib** [comment]
Closing this as a duplicate of #212
